File: src/main/java/architecture/ego_equipment/common/item/weapon/remote/gun/PistolEgoWeaponItem.java
package architecture.ego_equipment.common.item.weapon.remote.gun;

import architecture.ego_equipment.common.item.weapon.remote.RemoteEgoWeaponGeoItem;
import net.minecraft.resources.ResourceLocation;
import software.bernie.geckolib.model.GeoModel;

public class PistolEgoWeaponItem extends GunEgoWeaponItem {
	public PistolEgoWeaponItem(Properties itemProperties, Builder egoWeaponBuilder, GeoModel<RemoteEgoWeaponGeoItem> geoModel, GeoModel<RemoteEgoWeaponGeoItem> guiModel) {
		super(itemProperties, egoWeaponBuilder, geoModel, guiModel);
	}

	public PistolEgoWeaponItem(Properties itemProperties, Builder egoWeaponBuilder, ResourceLocation modPath) {
		super(itemProperties, egoWeaponBuilder, modPath);
	}
}

File: src/main/kotlin/architecture/ego_equipment/common/item/weapon/remote/gun/PistolEgoWeaponItem.kt
package architecture.ego_equipment.common.item.weapon.remote.gun

import architecture.ego_equipment.common.item.weapon.remote.RemoteEgoWeaponGeoItem
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.model.GeoModel

class PistolEgoWeaponItem : GunEgoWeaponItem {
	constructor(
		itemProperties: Properties,
		egoWeaponBuilder: Builder,
		geoModel: GeoModel<RemoteEgoWeaponGeoItem>,
		guiModel: GeoModel<RemoteEgoWeaponGeoItem>?
	) : super(itemProperties, egoWeaponBuilder, geoModel, guiModel)

	constructor(
		itemProperties: Properties,
		egoWeaponBuilder: Builder,
		modPath: ResourceLocation
	) : super(itemProperties, egoWeaponBuilder, modPath)
}
